### PR TITLE
FEATURE: EGL-367 - add the interactions drill down configuration

### PR DESCRIPTION
**Pull request by vannguyengd**
feat: add the interactions drill down configuration
_ adding the interactions configuration to the chart configuration panel 
_ turning off the drill down when the disableDrillDown is true

JIRA: EGL-367

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |
| **Compatibility matrix test commands - TIGER Backend**  |                                                            |
| `extended test - matrix-test <AIO_version>`             | Run integrated tests against AIO versions.                 |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

`<AIO_version>` in commands is used to start test with multiple AIO instances - each instance in triggered by one jenkins build

-   To run with `master` and `stable`, execute command `extended test - matrix-test master,stable` or `extended test - matrix-test latest`
-   To run with specific version,ex: `2.3.0` and `2.3.1`, execute command `extended test - matrix-test 2.3.0,2.3.1`
-   In case `<AIO_version>` is empty, read versions from file `compTigerVersions.txt` of this repo

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16732/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/18174/ : FAILURE in 2m 49s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16733/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/18175/ : SUCCESS in 11m 06s

**Comment by vannguyengd** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2508/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16735/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/1659/ : SUCCESS in 27m 12s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16752/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/18196/ : FAILURE in 8m 34s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16762/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/18203/ : FAILURE in 8m 25s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16763/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/18204/ : SUCCESS in 10m 58s

**Comment by vannguyengd** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2510/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16764/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/1667/ : UNSTABLE in 27m 21s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16765/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/18205/ : SUCCESS in 11m 21s

**Comment by vannguyengd** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2511/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16766/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/1668/ : SUCCESS in 27m 30s

File: libs/sdk-ui-ext/src/internal/components/configurationPanels/ConfigurationPanelContent.tsx [hunk 1]
```diff
@@ -99,4 +100,17 @@ export default abstract class ConfigurationPanelContent<
             />
         );
     }
+
+    protected renderInteractionsSection(): React.ReactNode {
+        const { featureFlags, pushData, properties, propertiesMeta } = this.props;
+
+        return featureFlags.enableAttributeHierarchies ? (
```
Comment (phong-nguyen-duy) [inline libs/sdk-ui-ext/src/internal/components/configurationPanels/ConfigurationPanelContent.tsx:107]: should check backend capabilities as well to ensure new UI is hidden in Bear

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16773/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/18214/ : SUCCESS in 11m 12s

**Comment by vannguyengd** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2514/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16781/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/1673/ : UNSTABLE in 27m 35s

**Comment by vannguyengd** [comment]
extended test - backstop

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16782/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/18219/ : SUCCESS in 11m 29s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2515/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16783/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/1674/ : UNSTABLE in 27m 11s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16784/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/18220/ : SUCCESS in 11m 17s

**Comment by vannguyengd** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2516/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16785/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/1675/ : SUCCESS in 27m 22s

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/DrillTargetType/DrillTargetTypeListItem.tsx [hunk 2]
```diff
@@ -25,12 +30,23 @@ const DrillTargetTypeListItem: React.FC<IDrillTargetTypeListItemProps> = ({
...
+                className={className}
+                title={item.title}
+                isSelected={isSelected}
+                onClick={handleClick}
+            />
+            {item.disabled && item.disableTooltipMessage ? (
+                <Bubble
+                    className="bubble-primary"
+                    alignPoints={[{ align: "cr cl" }]}
+                    arrowOffsets={{ "cr cl": [15, 0] }}
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/DrillTargetType/DrillTargetTypeListItem.tsx:44]: Please extract these two objects to constant outside of the component.

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/useDrillTargetTypeItems.ts [hunk 3]
```diff
@@ -47,6 +48,9 @@ export const useDrillTargetTypeItems = (): IDrillTargetType[] => {
             id: DRILL_TARGET_TYPE.DRILL_DOWN,
             title: intl.formatMessage(messages.drillDownConfig),
             disabled: true,
+            disableTooltipMessage: disableDrillDown
+                ? intl.formatMessage(messages.disableDrillDownToolTip)
+                : "",
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/useDrillTargetTypeItems.ts:53]: Undefined would be better than an empty string.

File: libs/sdk-ui-ext/src/internal/components/configurationControls/CheckboxControl.tsx [hunk 4]
```diff
@@ -55,7 +55,11 @@ class CheckboxControl extends React.Component<ICheckboxControlProps & WrappedCom
         const { valuePath, properties, pushData } = this.props;
 
         const clonedProperties = cloneDeep(properties);
-        set(clonedProperties, `controls.${valuePath}`, event.target.checked);
+        set(
+            clonedProperties,
+            `controls.${valuePath}`,
+            valuePath === "disableDrillDown" ? !event.target.checked : event.target.checked,
```
Comment (xMort) [inline libs/sdk-ui-ext/src/internal/components/configurationControls/CheckboxControl.tsx:61]: The component should imho not know about the valuePaths internals. 
I would rather add new optional prop to the component called `isValueInverted`, use this prop in this condition and in interaction section where the component is used set the prop to `true`. You don't need to update another occurrences of the Checkbox component as the prop would be optional and set to false by default for all these other instances.

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/InsightDrillConfigPanel/useInsightDrillConfigPanel.ts [hunk 5]
```diff
@@ -113,7 +114,10 @@ export const useInsightDrillConfigPanel = (props: IUseDrillConfigPanelProps) =>
         selectAllowMultipleInteractionsPerAttributeAndMeasure,
     );
     const widgetGlobalDrillDowns = useDashboardSelector(
-        selectGlobalDrillsDownAttributeHierarchyByWidgetRef(widgetRef),
+        selectGlobalDrillsDownAttributeHierarchyByWidgetRef(
+            widgetRef,
+            insight?.insight?.properties?.controls?.disableDrillDown,
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/InsightDrillConfigPanel/useInsightDrillConfigPanel.ts:119]: Does this selector signature need to be extended? You already have `widgetRef`. Can't you use it inside of the selector to get `insight` and inside of this selector get the prop? You would just change the selector and do not need to update all the other places where it is called.

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/ViewModeDashboardInsight/Insight/useDashboardInsightDrills.ts [hunk 6]
```diff
@@ -57,7 +58,9 @@ export const useDashboardInsightDrills = ({
     );
 
     const isInEditMode = useDashboardSelector(selectIsInEditMode);
-    const rawDrillableItems = useDashboardSelector(selectDrillableItemsByWidgetRef(widget.ref));
+    const rawDrillableItems = useDashboardSelector(
+        selectDrillableItemsByWidgetRef(widget.ref, disableDrillDownOnWidget),
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/presentation/widget/insight/ViewModeDashboardInsight/Insight/useDashboardInsightDrills.ts:62]: Does this selector signature need to be extended? You already have `widget.ref`. Can't you use it inside of the selector to get `insight` and inside of this selector get the prop? You would just change the selector and do not need to update all the other places where it is called.

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/InsightDrillConfigPanel/useInsightDrillConfigPanel.ts (hunk 5 above)
Comment (xMort) [reply libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/InsightDrillConfigPanel/useInsightDrillConfigPanel.ts:119]: Please check all the selector that you changed signature for if they can work with insight inside of them without changing the signature.

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/DrillTargetType/DrillTargetTypeListItem.tsx (hunk 2 above)
Comment (vannguyengd) [reply libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/DrillTargetType/DrillTargetTypeListItem.tsx]: I have updated this. Thank you!

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/useDrillTargetTypeItems.ts (hunk 3 above)
Comment (vannguyengd) [reply libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/useDrillTargetTypeItems.ts]: I have updated this. Thank you!

File: libs/sdk-ui-ext/src/internal/components/configurationControls/CheckboxControl.tsx (hunk 4 above)
Comment (vannguyengd) [reply libs/sdk-ui-ext/src/internal/components/configurationControls/CheckboxControl.tsx]: I have updated this. Thank you!

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/InsightDrillConfigPanel/useInsightDrillConfigPanel.ts (hunk 5 above)
Comment (vannguyengd) [reply libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/InsightDrillConfigPanel/useInsightDrillConfigPanel.ts]: I have updated this. Thank you!

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/ViewModeDashboardInsight/Insight/useDashboardInsightDrills.ts (hunk 6 above)
Comment (vannguyengd) [reply libs/sdk-ui-dashboard/src/presentation/widget/insight/ViewModeDashboardInsight/Insight/useDashboardInsightDrills.ts]: I have updated this. Thank you!

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16863/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/18293/ : SUCCESS in 11m 11s

**Review by xMort — APPROVED** [review]

**Comment by xMort** [comment]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR4319

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/3472/ : SUCCESS in 9m 40s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/18300/ : SUCCESS in 14m 01s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/1095/ : SUCCESS in 12m 00s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/4637/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/1352/ : FAILURE in 1m 21s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/1035/ : SUCCESS in 13m 27s

**Pull request merged**